### PR TITLE
fix: immediate return when dns server start err

### PR DESCRIPTION
**Pull request by beautytiger**
**What type of PR is this?**
 /kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: edgemesh/pkg/server/dns.go [hunk 1]
```diff
@@ -93,10 +93,11 @@ func startDnsServer() {
 		Port: 53,
 	}
 	udpConn, err := net.ListenUDP("udp", laddr)
-	defer udpConn.Close()
 	if err != nil {
 		klog.Errorf("Dns server Start error : %s", err)
+		return
 	}
+	defer udpConn.Close()
```
Comment (sids-b) [inline edgemesh/pkg/server/dns.go]: does this line cause a panic if it is above the if condition ?

Comment (beautytiger) [reply edgemesh/pkg/server/dns.go]: No, but the returned conn will be nil, it is no need to close it, like a nil file.
And the following process is also not needed, am I right?

**Comment by fisherxu** [comment]
Oh, just omit my comments, dnsConn is a global variable.
/lgtm

File: edgemesh/pkg/server/dns.go (hunk 1 above)
Comment (fisherxu) [reply edgemesh/pkg/server/dns.go]: Yes, if it is above the if condition, it will cause a panic.
e.g. https://play.golang.org/p/2SpVzVrnGjg

**Pull request merged**